### PR TITLE
PYTHON-2389 Add session support to find_raw_batches and aggregate_raw_batches

### PR DESCRIPTION
**Pull request by prashantmital**
~~ @ShaneHarvey what am I missing here? More tests? Edge cases? ~~

### CODE REVIEW

File: pymongo/collection.py [hunk 1]
```diff
@@ -1484,21 +1484,16 @@ def find_raw_batches(self, *args, **kwargs):
           >>> for batch in cursor:
           ...     print(bson.decode_all(batch))
 
-        .. note:: find_raw_batches does not support sessions or auto
-           encryption.
+        .. note:: find_raw_batches does not support auto encryption.
```
Comment (ShaneHarvey) [inline pymongo/collection.py:1487]: Do we have a ticket to implement encryption support now that we using OP_MSG?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
We should add tests to make sure these methods work with transactions, retryable reads, and snapshot reads as well.

File: pymongo/collection.py (hunk 1 above)
Comment (prashantmital) [reply pymongo/collection.py:1487]: Yes. I just linked it to this ticket as well as the ticket that implemented OP_MSG + raw_batches.

File: test/test_cursor.py [hunk 2]
```diff
@@ -1466,6 +1466,69 @@ def test_find_raw(self):
...
+            with session.start_transaction():
+                batches = list(client.db.test.find_raw_batches(
+                    session=session).sort('_id'))
+        self.assertEqual(1, len(batches))
+        self.assertEqual(docs, decode_all(batches[0]))
+
+        command = listener.results['started'][0].command
+        self.assertEqual(listener.results['started'][0].command_name, 'find')
+        self.assertEqual(command['$clusterTime'],
+                         decode_all(session.cluster_time.raw)[0])
```
Comment (ShaneHarvey) [inline test/test_cursor.py:1488]: Weird, `session.cluster_time` is a RawBSONDocument?

Comment (prashantmital) [reply test/test_cursor.py:1488]: Yes... :/

File: test/test_cursor.py [hunk 3]
```diff
@@ -1466,6 +1466,69 @@ def test_find_raw(self):
...
+        self.assertEqual(command['startTransaction'], True)
+        self.assertEqual(command['txnNumber'], 1)
+
+    @client_context.require_sessions
+    def test_find_raw_retryable_reads(self):
```
Comment (ShaneHarvey) [inline test/test_cursor.py:1493]: Needs a `@client_context.require_failCommand_fail_point`

File: test/test_cursor.py (hunk 2 above)
Comment (prashantmital) [reply test/test_cursor.py]: This is because of exhaust + OP_MSG changes.

**Review by ShaneHarvey — APPROVED** [review]
LGTM pending the outcome of the latest patch.
